### PR TITLE
feat: implement client.WithDefaultPageSize

### PR DESCRIPTION
**Pull request by QuentinBrosse**
Blocked by: #57

### CODE REVIEW

File: scw/client_option.go [hunk 1]
```diff
@@ -121,3 +121,12 @@ func WithDefaultZone(zone utils.Zone) ClientOption {
 		s.defaultZone = zone
 	}
 }
+
+// WithDefaultPageSize client option overrides the default page size of the SDK.
+//
+// It will be used as the default value of the page_size field in all requests made with this client.
+func WithDefaultPageSize(pageSize int32) ClientOption {
+	return func(s *settings) {
+		s.defaultPageSize = pageSize
```
Comment (jerome-quere) [inline scw/client_option.go]: should check that page size is not 0

File: scw/client.go [hunk 2]
```diff
@@ -89,6 +91,13 @@ func (c *Client) GetDefaultZone() utils.Zone {
 	return c.defaultZone
 }
 
+// GetDefaultPageSize return the default page size of the client.
+// This value can be set from the client option
```
Comment (DenBeke) [inline scw/client.go]: can be set **in** the client option

File: scw/client_option.go (hunk 1 above)
Comment (kindermoumoute) [reply scw/client_option.go]: The check should not happen here but in [`settings.validate()`](https://github.com/scaleway/scaleway-sdk-go/blob/master/scw/settings.go#L33)

File: scw/client.go [hunk 3]
```diff
@@ -98,6 +100,13 @@ func (c *Client) GetDefaultZone() (utils.Zone, bool) {
 	return utils.Zone(""), false
 }
 
+// GetDefaultPageSize return the default page size of the client.
+// This value can be set in the client option
+// WithDefaultPageSize(). Be aware this value can be empty.
+func (c *Client) GetDefaultPageSize() (int32, bool) {
+	return *c.defaultPageSize, c.defaultPageSize != nil
```
Comment (kindermoumoute) [inline scw/client.go]: will panic if c.defaultPageSize is nil

**Pull request merged**